include/superior_mysqlpp/config.hpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Author: Jiří Kozlovský
+ */
+
+#pragma once
+
+#include <string>
+
+namespace SuperiorMySqlpp
+{
+    struct SslConfiguration
+    {
+        const char* keyPath = nullptr;
+        const char* certificatePath = nullptr;
+        const char* certificationAuthorityPath = nullptr;
+        const char* trustedCertificateDirPath = nullptr;
+        const char* allowableCiphers = nullptr;
+    };
+
+    class ConnectionConfig {
+        private:
+            ConnectionConfig(const bool _hasSsl, const struct SslConfiguration& _sslConfig, const std::string& _database, const std::string& _user, const std::string& _password,
+                    const std::string& _target, const std::uint16_t _port=0) : 
+                hasSsl{_hasSsl},
+                sslConfig{_sslConfig},
+                usingSocket{_port == 0},
+                database{_database},
+                user{_user},
+                password{_password},
+                target{_target},
+                port{_port}
+            { }
+
+        public:
+
+            const bool hasSsl;
+            const struct SslConfiguration sslConfig;
```

Comment:
Please use `SuperiorMySqlpp::Optional<SslConfiguration>` instead of struct and bool